source/fundamentals/read-write-configuration.txt

```diff
@@ -0,0 +1,205 @@
+--------
+
+In this guide, you can learn how to use the **write concern**, **read concern**,
+and **read preference** configurations to modify the way that MongoDB runs
+create, read, update, and delete (CRUD) operations on replica sets.
+
+You can set these configurations at the following levels:
+
+1. Client, which sets the default for all operation executions unless overridden
+#. Transaction
+#. Database
+#. Collection
+
+This list is in increasing order of precedence. For example, if you set
+read concerns at both the client and the database levels, the read
+concern specified at the database level overrides the read concern at the
+client level.
+
+Write Concern
+-------------
+
+Write concern specifies the level of acknowledgement requested from MongoDB for
+write operations before the operation successfully returns. Operations that
+don't specify an explicit write concern inherit the global default write concern
+setting.
+
+You can set the write concern by setting the ``WriteConcern`` option on a
+``MongoClientSettings``, ``MongoDatabaseSettings``, or ``MongoCollectionSettings`` object,
+or by using the ``WithWriteConcern()`` method of a client, database, or collection instance.
+
+The ``WriteConcern`` option and ``WithWriteConcern()`` method accept a
+``WriteConcern`` instance as a parameter. You can specify the write concern by 
+using one of the following values:
+
+- ``WriteConcern.Acknowledged``: The write operation returns after the operation
+  is written to memory.
+- ``WriteConcern.W1``: The write operation returns after only the primary node acknowledges
+  the write operation, without waiting for acknowledgement from secondary nodes.
+- ``WriteConcern.W2``: The write operation returns after the primary node and
+  at least one secondary node acknowledge the write operation.
+- ``WriteConcern.W3``: The write operation returns after the primary node and
+  at least two secondary nodes acknowledge the write operation.
+- ``WriteConcern.WMajority``: The write operation returns after a majority of the
+  replica set members acknowledge the write operation.
+- ``WriteConcern.Unacknowledged``: The write operation returns after the primary
+  node processes the write operation.
+
+The following example sets the write concern to ``WriteConcern.WMajority`` for an instance of
+``MongoClient``:
+
+.. literalinclude:: /includes/fundamentals/code-examples/ReplicaSetConfigs.cs
+   :start-after: start-write-concern-client
+   :end-before: end-write-concern-client
+   :language: csharp
+   :dedent:
+
+The following example sets the write concern to ``WriteConcern.WMajority`` for a collection:
+
+.. literalinclude:: /includes/fundamentals/code-examples/ReplicaSetConfigs.cs
+   :start-after: start-write-concern-collection
+   :end-before: end-write-concern-collection
+   :language: csharp
+   :dedent:
+
+.. note:: Collections and Databases are Immutable
+
+   ``IMongoDatabase`` and ``IMongoCollection`` instances are immutable. When you
```

Comment:
`IMongoClient` instances are also immutable as you can't modify settings after creating them. If you need to apply different settings to an `IMongoClient`, `IMongoDatabase`, or `IMongoCollection`, you must use one of the `With*` methods. These methods are designed to always return new instances with the specified settings. I recommend updating this note to reflect this information.